test/kaniko_task_test.go

```diff
@@ -89,12 +116,40 @@ func getTask(namespace string, t *testing.T) *v1alpha1.Task {
 					Name:  "kaniko",
 					Image: "gcr.io/kaniko-project/executor",
 					Args: []string{"--dockerfile=/workspace/Dockerfile",
-						fmt.Sprintf("--destination=%s/kanikotasktest", dockerRepo),
+						fmt.Sprintf("--destination=%s", repo),
 					},
 				}},
 			},
 		},
 	}
+	if withSecretConfig {
+		step := task.Spec.BuildSpec.Steps[0]
```

Comment:
instead of indexing into `Steps`, one thing you could do is to hold the `Step` in a variable, .e.g.:

```
step: = corev1.Container{{...}}
...
BuildSpec:..... Steps: []corev1.Container{{step}}...
...
if withSecretConfig {
   step.VolumeMounts...
```

I think this is slightly more flexible than depending on the Step order